src/remote.c

```diff
@@ -670,21 +670,32 @@ int git_remote_set_pushurl(git_repository *repo, const char *remote, const char*
 	return set_url(repo, remote, CONFIG_PUSHURL_FMT, url);
 }
 
-const char* git_remote__urlfordirection(git_remote *remote, int direction)
+static int git_remote__urlresolve(const char *url, char **resolved_url, int direction, const git_remote_callbacks *callbacks)
 {
-	assert(remote);
+	if (callbacks && callbacks->resolve_url) {
+		return callbacks->resolve_url(url, resolved_url, direction, callbacks->payload);
+	}
```

Comment:
No need for those braces. Edit: I see there's precedent though in the other functions. So you're free to do as you wish ;)